polars/polars-core/src/chunked_array/binary/encoding.rs

```diff
@@ -0,0 +1,35 @@
+use {base64, hex};
+
+use crate::prelude::*;
+
+impl BinaryChunked {
+    pub fn hex_decode(&self, strict: Option<bool>) -> PolarsResult<BinaryChunked> {
+        let ca = self.apply_on_opt(|e| e.and_then(|s| hex::decode(s).map(Into::into).ok()));
+
+        if strict.unwrap_or(false) && (ca.null_count() != self.null_count()) {
+            Err(PolarsError::ComputeError("Unable to decode inputs".into()))
+        } else {
+            Ok(ca)
+        }
+    }
+
+    #[must_use]
+    pub fn hex_encode(&self) -> BinaryChunked {
+        self.apply(|s| hex::encode(s).into_bytes().into())
+    }
+
+    pub fn base64_decode(&self, strict: Option<bool>) -> PolarsResult<BinaryChunked> {
```

Comment:
not really, I can encode any byte string into Utf8 using base64 so if I am decoding base64 there is no guarantee it will be valid utf8, you can see it in base64 library API as well. same thing with hex.

Reply:
if you mean the output of encode, this is old code it is fixed after last commit